### PR TITLE
Train station ruin

### PR DESCRIPTION
**Pull request by immortalartisan**
## About The Pull Request

Adds in a new lavaland ruin that is meant to be a train station overrun with goliaths and some legion around to indicate what were corpses before 
SSDM Screenshot with no area
image
SSDM with area 
image
4d6f32b6fbc7)
in game screenshot
image

## Why It's Good For The Game

More ruins for lavaland!

## Changelog

:cl:
add: New lavaland ruin Train station
/:cl:

### CODE REVIEW

File: code/datums/ruins/lavaland.dm [hunk 1]
```diff
@@ -142,3 +142,9 @@
 	id = "codelab"
 	description = "A Nanotrasen genetic research facility, abandoned and ripe for looting. Whats that goo over there?"
 	suffix = "lavaland_surface_codelab.dmm"
+/datum/map_template/ruin/lavaland/codelab
+	name = "Train station"
+	id = "train_station"
+	description = "A Old and overun train station who built it and what were they moving?"
```
Comment (meemofcourse) [inline code/datums/ruins/lavaland.dm]:
```suggestion
	description = "An old and abandoned train station. Who built it, and what were they moving?"
```

**Review by meemofcourse — CHANGES_REQUESTED** [review]

File: code/datums/ruins/lavaland.dm (hunk 1 above)
Comment (immortalartisan) [reply code/datums/ruins/lavaland.dm]: is that all that is requested?

Comment (meemofcourse) [reply code/datums/ruins/lavaland.dm]: you also have to address the discord changes the other maptainer gave you.

Comment (immortalartisan) [reply code/datums/ruins/lavaland.dm]: > you also have to address the discord changes the other maptainer gave you.

I do believe that with the new commit ive addressed those changes now

**Review by Latentish — CHANGES_REQUESTED** [review]
- Radiation causes server lag, do something else with that area.
- The train waiting platforms with the ripley is rather bland, please give it some more flavoring. you can do this with added decal work, debris, and clutter.
- I don't particularly understand the bedding area, I'd likely prefer you change this area up. Also try to avoid separating rooms with thindows.
- The side area with the smes needs more flavor. select a tiling and put some decal work/desbris/clutter
- I'd prefer you used a different tiling than concrete for this. For the outside area it's more acceptable, however the building being one grey rectangle of cement just doesn't look good.
- I don't quite understand these scrubbers you've placed around.
- For a trainstation they're packing some serious heat, I think the loot pool here could be reconsidered to make more sense for a train station.

Overall this is a nice start to a ruin, I like your train and the surrounding environment. A little work on the platforms, redoing the cement building a good bit, and adding more detailing to the platforms will make this a very very excellent ruin.

**Review by Latentish — CHANGES_REQUESTED** [review]
You need to add a ruin catalogue entry.

**Comment by immortalartisan** [comment]
> You need to add a ruin catalogue entry.

there is already one?

**Comment by immortalartisan** [comment]
> * Radiation causes server lag, do something else with that area.
> * The train waiting platforms with the ripley is rather bland, please give it some more flavoring. you can do this with added decal work, debris, and clutter.
> * I don't particularly understand the bedding area, I'd likely prefer you change this area up. Also try to avoid separating rooms with thindows.
> * The side area with the smes needs more flavor. select a tiling and put some decal work/desbris/clutter
> * I'd prefer you used a different tiling than concrete for this. For the outside area it's more acceptable, however the building being one grey rectangle of cement just doesn't look good.
> * I don't quite understand these scrubbers you've placed around.
> * For a trainstation they're packing some serious heat, I think the loot pool here could be reconsidered to make more sense for a train station.
> 
> Overall this is a nice start to a ruin, I like your train and the surrounding environment. A little work on the platforms, redoing the cement building a good bit, and adding more detailing to the platforms will make this a very very excellent ruin.

Right so the main idea is that the structure is a barracks set up at some point in the distant past ... weather it was to hold the planet or fight off fauna im leaving that in the air. that is my main explanation for why this train has so much heat

**Review by Latentish — CHANGES_REQUESTED** [review]
image
I'd like you to put some carpeting or other varied tiling in here to break it up a bit. I'd recommend putting it under one of the desks and wrapping your wood border decal around it.

image
Vary up the tiling here. You can use certain tiling to imply electronics being hooked into the floor underneath, or imply where wires are being directed.

image
I want you to rethink this platform. I don't particularly like that you're using engine flooring for it. The way you're using gore decals and a bit of clutter is alright, but the platform is still largely empty. The way you're using hazard stripe decals vertically here overlaps the other decals here which make them look awkward. You should think about how all these decals look together and if they're awkward at all. I would recommend trying to use things like railings, multiple floor tiles other than open plating, using some benches, maybe a broken console along the wall for flavor. Things like that can spice it up. But it should look a bit more refined than this. The same goes for the bottom platform.

image
So I like your flavor here, I think the way you've used crystals and the black dirt tells a nice story for radiation. But without something there but clothes, it just looks dilapidated and empty for little reason.  I'd just like you to add a few things to the floor here really, fill it up a bit. Doesn't have to be gamer look just some objects for flavor, like the suit.

Overall this is a solid improvement.

**Comment by immortalartisan** [comment]
image
those are not null rods their just icon edited and renamed normal metal rods

**Comment by Latentish** [comment]
I'm a little concerned about how players will handle the crystal legion in a confined space. So what I'm gonna do is tm this and get back to you.

**Comment by Latentish** [comment]
I want you to test this on local. Clear it out with on your own. Like spawn yourself in with some gear and do it.
You will need to fix your ruin name, because the name of the .dmm and the filepath you put in do not match. They are case sensitive. which means this ruin does not spawn.

File: code/datums/ruins/lavaland.dm [hunk 2]
```diff
@@ -126,8 +126,15 @@
 	description = "A Nanotrasen genetic research facility, abandoned and ripe for looting. Whats that goo over there?"
 	suffix = "lavaland_surface_codelab.dmm"
 
+/datum/map_template/ruin/lavaland/codelab
```
Comment (Latentish) [inline code/datums/ruins/lavaland.dm]: You're using the /codelab here, so you're effectively being overwritten by the actual ruin that came fist.

File: _maps/map_catalogue.txt [hunk 3]
```diff
@@ -194,6 +194,10 @@ Find the key for using this catalogue in "map_catalogue_key.txt"
 		Size = (x = 56)(y = 35)(z = 1)
 		Tags = "Minor Combat Challenge", "Medium Loot", "Shelter", "Hazardous"
 
+		File Name = "_maps\RandomRuins\LavaRuins\lavaland_surface_trainstaion.dmm"
```
Comment (Latentish) [inline _maps/map_catalogue.txt]: trainstaion. This should be lavaland_surface_train_station.dmm

File: code/datums/ruins/lavaland.dm [hunk 4]
```diff
@@ -126,8 +126,15 @@
...
+	name = "Train station"
```
Comment (Latentish) [inline code/datums/ruins/lavaland.dm]: Normalize your casing here. "Train Station"

File: code/datums/ruins/lavaland.dm [hunk 5]
```diff
@@ -126,8 +126,15 @@
...
+	id = "train_station"
+	description = "An old and abandoned train station. Who built it, and what were they moving?"
```
Comment (Latentish) [inline code/datums/ruins/lavaland.dm]: I would recommend asking a lorema for help with a better description here. This is very lackluster.

File: code/game/area/areas/ruins/lavaland.dm [hunk 6]
```diff
@@ -99,6 +99,18 @@
 	name = "Nanotrasen Genetic Research Maintenance"
 	icon_state = "dk_yellow"
 
+/area/ruin/unpowered/train_station/train
```
Comment (Latentish) [inline code/game/area/areas/ruins/lavaland.dm]: You don't need 3 new areas for this ruin. Try cutting down to like, 1. You could very easily be using /ruin/unpowered for the main barracks area, /external for the platform, and 1 unique one for the train. Though Honestly you could just not have any if the power set up here isn't actually important. (This doesn't seem like a power puzzle, and unless that's a magical smes this won't be powered normally.)

**Review by Latentish — CHANGES_REQUESTED** [review]
correct your dmm file name. it should be in line with all the others. Double check your spelling, cases, and use of underlines. Your map should be named "lavaland_surface_train_station.dmm"

image
This is going to mulch people. change the crystal legion out for like, 2 normal legions, and make the legion on the platform a crystal legion.

image
None of these middle rooms have lights. Also trim down on the mech loot here. They're getting a ripley already, don't give them the super miner plasma cutter.

image
Put some minor stuff in here for people to grabby hands at. Nothing big.

image
Give a reason to go to these rooms. Change up the food, add laser tag guns in the lockers or pj's or other fluff stuff.

image
This room is very bland. I would recommend rethinking it and the armory area in there. Also do not do that to rechargers. Do not put them in walls.

image
This seating is very strange. Just make them facing south backed against the pillars.

image
image
You have 4 pacmen and 2 smes on this ruin. You do not need to make this train electronically viable. I would recommend replaces these with garbage circuits and machine frames, keeping like, 1 smes and 1 pacman up in the cargo area up top.

Moreover, what are these crates? Are these the crates that spawn with stuff in them, or did you replace the sprite of the basic crate in sdmm? Because if these are the regular crates, you're giving a lot more look than you think you are.

**Comment by Latentish** [comment]
Ok nevermind don't use crystal mobs at all in this ruin.

**Comment by Latentish** [comment]
Please add an in game screenshot to your pr.

**Review by Latentish — CHANGES_REQUESTED** [review]
image
fence int he wall here.

image
Make this either an airlock or a holofield with shutters.

image
All these outside tiles will be active on load due to being next to a different atmos tile. The platform and train either need to be enclosed without lavaland atmos tiles in them, or you need to make subtypes of turfs for use on lavaplanets.

image
Wires do not work through walls, make this a window like so.

image
These areas are funky. make it just that walkway in the middle to be the train_station area.

image
image
This place is outside, give it an outside area.

**Comment by Latentish** [comment]
closing on contrib request

**Pull request closed without merging**